examples/example-libevent-pubsub.c

```diff
@@ -0,0 +1,169 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <signal.h>
+
+#include <assert.h>
+#include <hiredis.h>
+#include <async.h>
+#include <adapters/libevent.h>
+
+void printReplyInternal(const redisReply* r) {
+    switch (r->type) {
+    case REDIS_REPLY_INTEGER:
+        printf("(integer %lld)", r->integer);
+        break;
+    case REDIS_REPLY_DOUBLE:
+        printf("(double %f)", r->dval);
+        break;
+    case REDIS_REPLY_ERROR:
+        printf("(error %s)", r->str);
+        break;
+    case REDIS_REPLY_STATUS:
+        printf("(status %s)", r->str);
+        break;
+    case REDIS_REPLY_STRING:
+        printf("(string %s)", r->str);
+        break;
+    case REDIS_REPLY_VERB:
+        printf("(verb %s)", r->str);
+        break;
+    case REDIS_REPLY_ARRAY:
+        printf("(array %zu", r->elements);
+        for (size_t i = 0; i < r->elements; ++i) {
+            putchar(' ');
+            printReplyInternal(r->element[i]);
+        }
+        putchar(')');
+        break;
+    default:
+        printf("(?%d)", r->type);
+        break;
+    }
+}
+
+void printReply(const redisReply* r) {
+    printReplyInternal(r);
+    putchar('\n');
+}
+
+void getCallback(redisAsyncContext *c, void *r, void *privdata) {
+    redisReply *reply = r;
+    if (reply == NULL) {
+        if (c->errstr) {
+            printf("errstr: %s\n", c->errstr);
+        }
+        return;
+    }
+    printf("argv[%s]: %s\n", (char*)privdata, reply->str);
+}
+
+void getFinalizer(redisAsyncContext *c, void *privdata) {
+    printf("Get finalizer called\n");
```

Comment:
Just to avoid the warning

```suggestion
    (void)privdata;
    printf("Get finalizer called\n");
```